src/Form/Project/ProcesVerbalSubType.php

```diff
@@ -11,24 +11,45 @@
 
 namespace App\Form\Project;
 
+use App\Entity\Project\Etude;
+use App\Entity\Project\Phase;
 use App\Entity\Project\ProcesVerbal;
-use Symfony\Component\Form\Extension\Core\Type\IntegerType;
+use App\Repository\Project\PhaseRepository;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
 use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
 class ProcesVerbalSubType extends DocTypeType
 {
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
-        $phaseNum = $options['phases'];
-        if ('pvi' == $options['type']) {
-            $builder->add(
-                'phaseID',
-                IntegerType::class,
-                ['label' => 'Phases concernées', 'required' => false, 'attr' => ['min' => '1', 'max' => $phaseNum]]
-            );
+        if (!isset($options['etude']) || !($options['etude'] instanceof Etude)) {
+            throw new \LogicException('A ProcesVerbalSubType can\'t be build without associated Etude object.');
         }
 
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) use ($options) {
+            /** @var ProcesVerbal $proces */
+            $proces = $event->getData();
+            $form = $event->getForm();
+
+            $form->add('phases', EntityType::class, [
+                'class' => Phase::class,
+                'query_builder' => function (PhaseRepository $pr) use ($options, $proces) {
+                    if (isset($proces)) {
+                        return $pr->getByEtudeAndPv($options['etude'], $proces);
+                    } else {
+                        return $pr->getByEtudeAndPv($options['etude']);
+                    }
```

Comment:
```suggestion
                    if (isset($proces)) {
                        return $pr->getByEtudeAndPv($options['etude'], $proces);
                    }
                    return $pr->getByEtudeAndPv($options['etude']);
```
Le else n'est pas nécessaire quand le if se termine par un return. Le supprimer améliore la lisibilité.

J'ai conscience que mes derniers commentaires sont assez pointilleux. Mais ils sont basés sur ce qui est déjà en place sur Jeyser et ce sont des règles que tu retrouveras souvent dans le monde professionnel. C'est en ça que l'open-source est une bonne école pour tout ce qui est qualité de code.

Reply:
Yes, c'est mon code qui n'est pas très propre surtout, désolé pour ça